### PR TITLE
Shorten file names

### PR DESCRIPTION
**Pull request by sdquiring**
Adding a mechanism which will allow kernel file names to be shortened by hashing the rest of the kernel name beyond a certain point.

Note that the kernel name does not change.

Also creating a new replacement kernel mechanism which looks inside the files for the kernel name so that replacement kernels will still work.

I was able to build rocBLAS' kernels with `--no-merge-files` with these changes.

### CODE REVIEW

**Comment by pfultz2** [comment]
I ran this in miopentensile and I still get filename too long:

```
objcopy:/tmp/tmp.PLcRI3TlUY/Cijk_Ailk_Bjlk_DB_MT64x128x4_SN_APM1_AF0EM1_AF1EM1_AMAS3_ASEM1_BL1_DTL0_EPS1_FL1_GRVW2_GSU1_ISA908_IU1_K1_KLA_LPA0_LPB0_LDL1_NLCA1_NLCB1_ONLL1_PBD0_PK0_PGR1_PLR1_RK0_SU0_SNLL1_TT4_8_USFGRO0_VAW1_VW2_WG16_16_1_WGM1.eaab1092034a5b97b14ecc7b26046912.kernel.bc: File name too long
```

It does show the hash but the name is still too long.

File: Tensile/Common.py [hunk 1]
```diff
@@ -156,6 +156,7 @@
 globalParameters["MaxDepthU"] = 256               # max DepthU value to allow
 globalParameters["ShortNames"] = False            # on windows kernel names can get too long; =True will convert solution/kernel names to serial ids
 globalParameters["MergeFiles"] = True             # F=store every solution and kernel in separate file; T=store all solutions in single file
+globalParameters["MaxFileName"] = 128 # If a file name would be longer than this, shorten it with a hash.
```
Comment (pfultz2) [inline Tensile/Common.py]: How about setting this to `64`? It seems like `128` will still make the name too long.

Comment (zaliu) [reply Tensile/Common.py]: rocBLAS build with this PR has a maximum file name length of 143.  In a docker container the limit seems to be around 224.  If any utility adds filename overhead as large as 80, that utility needs work.  Back to your issue.  I'm sure you can set your own MaxFileName to 64.  The default value of 128 appears to be quite sensible.

Comment (zaliu) [reply Tensile/Common.py]: rocBLAS build with this PR has a maximum file name length of 143.  In a docker container the limit seems to be around 224.  If any utility adds filename overhead as large as 80, that utility needs work.  Back to your issue.  I'm sure you can set your own MaxFileName to 64.  The default value of 128 appears to be quite sensible.

Comment (zaliu) [reply Tensile/Common.py]: A rocBLAS build with this PR has a maximum file name length of 143.  In a docker container the limit seems to be around 224.  If any utility adds filename overhead as large as 80, that utility needs work.  Back to your issue.  I'm sure you can set your own MaxFileName to 64.  The default value of 128 appears to be quite sensible.

Comment (zaliu) [reply Tensile/Common.py]: A rocBLAS build with this PR has a maximum file name length of 143.  In a docker container the limit seems to be around 224.  If any utility adds filename overhead as large as 80, that utility needs work.  Back to your issue.  I'm sure you can set your own MaxFileName to 64.  The default value of 128 appears to be quite sensible.

Comment (pfultz2) [reply Tensile/Common.py]: I am seeing a max file length of 277, which is too large. How do I set my own `MaxFileName` without forking Tensile?

**Comment by pfultz2** [comment]
Ok, so after getting the forked repo, I am no longer seeing too long of filenames.

**Pull request merged**